### PR TITLE
Implemented Main address list display and started basic modal pushing

### PR DESCRIPTION
**Pull request by TheResinger**
This PR sets up the main display logic for the address list. 

### To-Do

Still need to set up the front end API to connect with the back ends api routes to pull data, for the time being there is just some generated hard coded data.

The buttons for the users in the address list are set to call the modal window, but the modal isint currently set up to display the data passed to them. This is still a WIP.

### CODE REVIEW

File: frontend/address-book/src/components/Main.jsx [hunk 1]
```diff
@@ -1,13 +1,87 @@
...
+                    {customerState.filter(customer => {
+                        if (!currentSearchValue){
+                            return customer;
+                        } else if (customer.Name.toLowerCase().includes(currentSearchValue.toLowerCase())) {
+                            return customer;
+                        }
+                    }).map(customer => (
+                        <div className="customerCard" id={customer._id} key={customer._id}>
+                            <div className="customerCardBody">
+                                <div className="customerCardName">{customer.firstName} {customer.lastName}</div>
```
Comment (JEub) [inline frontend/address-book/src/components/Main.jsx:55]: Want to test this, and/or see a demo prior to merging due to the single brace.

File: frontend/address-book/src/states.js [hunk 2]
```diff
@@ -7,4 +7,19 @@ export const searchInputState = atom({
 export const currentSearchTerm= atom({
```
Comment (JEub) [inline frontend/address-book/src/states.js]: Can you fix your friend's formatting here and add a space after `atom`?

**Review by JEub — CHANGES_REQUESTED** [review]

File: frontend/address-book/src/states.js (hunk 2 above)
Comment (TheResinger) [reply frontend/address-book/src/states.js]: This has been updated in the latest push.

File: frontend/address-book/src/components/Main.jsx (hunk 1 above)
Comment (TheResinger) [reply frontend/address-book/src/components/Main.jsx:55]: [Here is a video showing off how it works](https://youtu.be/Jb-RXazq42c), I noticed that a subvariable was missed when refactoring for the modal support, so I also went and updated that as well.

**Pull request closed without merging**